CONTRIBUTING.md

```diff
@@ -1,16 +1,160 @@
 # How to contribute
 
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+<!-- generated with [DocToc](https://github.com/thlorenz/doctoc)* -->
+
+- [How to contribute to Matomo core?](#how-to-contribute-to-matomo-core)
+- [How to submit a bug report or suggest a feature?](#how-to-submit-a-bug-report-or-suggest-a-feature)
+- [How to suggest improvements to translations?](#how-to-suggest-improvements-to-translations)
+- [How to submit code improvements via pull requests?](#how-to-submit-code-improvements-via-pull-requests)
+- [Docker-based development environment](#docker-based-development-environment)
+  - [Capabilities](#capabilities)
+  - [Prerequisites](#prerequisites)
+  - [Matomo admin user credentials](#matomo-admin-user-credentials)
+  - [Managing the virtual environment](#managing-the-virtual-environment)
+    - [Running](#running)
+    - [Stopping](#stopping)
+    - [Deleting](#deleting)
+  - [How to run some CLI in Matomo container](#how-to-run-some-cli-in-matomo-container)
+  - [How to update the Docker development environment to a new version of Matomo](#how-to-update-the-docker-development-environment-to-a-new-version-of-matomo)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+
+<a name="how-to-contribute-to-matomo-core"></a>
+## How to contribute to Matomo core?
+
 Great to have you here! Read the following guide on our developer zone to learn how you can help make this project better!
 
 https://developer.matomo.org/guides/contributing-to-piwik-core
 
+<a name="how-to-submit-a-bug-report-or-suggest-a-feature"></a>
 ## How to submit a bug report or suggest a feature?
 Please read the recommendations on writing a good [bug report](https://developer.matomo.org/guides/core-team-workflow#submitting-a-bug-report) or [feature request](https://developer.matomo.org/guides/core-team-workflow#submitting-a-feature-request).
 
+<a name="how-to-suggest-improvements-to-translations"></a>
 ## How to suggest improvements to translations?
 
 You can help improve translations in Matomo, please read [contribute to translations](https://github.com/matomo-org/piwik/blob/master/lang/README.md).
 
+<a name="how-to-submit-code-improvements-via-pull-requests"></a>
 ## How to submit code improvements via pull requests?
 
 You can help contribute to Matomo codebase via Pull Requests, see [Contributing to Matomo core](https://developer.matomo.org/guides/contributing-to-piwik-core)
+
+
+<a name="docker-based-development-environment"></a>
```

Comment:
Can we instead maybe simply link to our official docker? https://github.com/matomo-org/docker